mlrun/projects/project.py

```diff
@@ -763,6 +765,19 @@ def get_code_path(self):
         """Get the path to the code root/workdir"""
         return path.join(self.context, self.workdir or self.subpath or "")
 
+    def _replace_default_image_in_enriched_functions(self, previous_image, new_image):
+        """
+        Set a new project-default-image in functions that were already enriched.
+        """
+        if previous_image == new_image:
+            return
+        for key in self._function_objects:
+            function = self._function_objects[key]
+            if hasattr(function, "_enriched_image") and getattr(
+                function, "_enriched_image", False
+            ):
```

Comment:
why hasattr/getattr and not simply `function._enriched_image`? why are we uncertain whether the property is set/unset here? once you override the function image, set the enriched to true? and also, should you check if _not_ enriched and then override the image?
using `setattr` is too implicit and make it hard to track, could be better mechanism for such cases

Reply:
@liranbg - This property (`_enriched_image`) is specific to the use-case of holding functions in the project object, and while it is a property of the function, it is only used by the project. Therefore it's not mandatory to have this property on the function at all times. OTOH, I see your point and there really isn't too much damage in converting this to a full-time property of the object, so I'll change.
As for why checking for already enriched - this function is called in the flow where a previous default image was used and is replaced, therefore we only replace on functions that were already enriched with the previous version of the default image.